lib/logstash/inputs/file.rb

```diff
@@ -210,126 +261,127 @@ def register
       end
 
       #pick SINCEDB_DIR if available, otherwise use HOME
-      sincedb_dir = ENV["SINCEDB_DIR"] || ENV["HOME"]
+      sincedb_dir = Pathname.new(ENV["SINCEDB_DIR"] || ENV["HOME"])
 
       # Join by ',' to make it easy for folks to know their own sincedb
       # generated path (vs, say, inspecting the @path array)
-      @sincedb_path = File.join(sincedb_dir, ".sincedb_" + Digest::MD5.hexdigest(@path.join(",")))
+      @sincedb_path = sincedb_dir.join(".sincedb_" + Digest::MD5.hexdigest(@path.join(",")))
```

Comment:
I notice code duplication with the sincedb filename generation, we should DRY it up